File: Difficulty: Easy/Reverse Words/reverse-words.java
//{ Driver Code Starts
import java.util.*;
import java.lang.*;
import java.io.*;
class GFG {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t > 0) {
            String s = sc.next();
            Solution obj = new Solution();
            System.out.println(obj.reverseWords(s));
            t--;
        }
    }
}

// } Driver Code Ends



class Solution 
{
    //Function to reverse words in a given string.
    String reverseWords(String S)
    {
        // code here 
        int n = S.length();
        int i = 0;
        char[] chstr = S.toCharArray();
        // System.out.println(chstr);
        for(; i < n; i++)
        {
            if(S.charAt(i) != '.')
            {
                break;
            }
        }
        
        StringBuilder sb = new StringBuilder();
        String result = "";
        
        for(; i < n; i++)
        {
            if(chstr[i] != '.')
            {
                sb.append(chstr[i]);
            }
            else
            {
                if(sb.length() != 0)
                {
                    sb.reverse();
                    result = result + sb.toString() + '.';
                    sb.setLength(0);
                }
            }
        }
        
        if(sb.length() != 0)
        {
            sb.reverse();
            result += sb.toString();
            sb.setLength(0);
        }
        
        sb.append(result);
        sb.reverse();
        
        return sb.toString();
    }
}

File: Difficulty: Easy/Reverse Words/reverse-words.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends

class Solution {
  public:
    // Function to reverse words in a given string.
    string reverseWords(string str) {
        // code here
          vector<string> vr;
        int start = 0;
        int count = 0;
        for(int i = 0; i<str.size(); i++){
            count+=1;
            if(str[i] == '.'){
                string temp = str.substr(start,count-1);
                vr.push_back(temp);
                start = i+1;
                count = 0;
            }
        }
        vr.push_back(str.substr(start));
        string ans;
        int n = vr.size();
        for(int i = n-1; i>0; i--){
            ans = ans + vr[i] + '.';
        }
        ans += vr[0];
        return ans;
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        string s;
        cin >> s;
        Solution obj;
        cout << obj.reverseWords(s) << endl;
    }
}
// } Driver Code Ends